### DISCUSSION TITLE
Issue with generateProto Gradle task in multi-subproject build

### DISCUSSION THREAD

**Discussion by TheBroach** [quarkusio/quarkus#43492] [Q&A] [answered]
Hi!
I am building a "blueprint" project to use for new projects with Quarkus + Kotlin + Gradle multi-subproject build + Ports & Adapters design pattern.

[Here is my example project](https://github.com/TheBroach/quarkus-gradle-cdi-mcve).

At the moment I am trying to understand how to use the Quarkus Gradle plugin. Currently, it is only applied in the "application" subproject.

I have recently started working with gRPC and wanted to include that in this project. A probable scenario would be that the domain needs to call a remote service, so I started to set up a gRPC client in the "driven-adapters" subproject. I found in the official Quarkus gRPC guide that I should configure the Quarkus Gradle plugin to be aware of my special directory for proto files. However, after just pasting in a simple "Hello World" proto file and trying to build, I get an error that the `protoc` compiler is not found. The Quarkus Gradle plugin should be fetching that, so maybe I am missing some configuration?

Thanks!

[a comment by an automated account is not shown]

**Reply by TheBroach** [reply]
No thought from @geoand or @glefloch ? Maybe @cescoffier has some gRPC-related insight?

**Reply by cescoffier** [reply]
The gRPc codegen task should be independent from the build tools. That being said, I'm only using it with Maven. 

Note that it downloads protoc into the build directory. Maybe in the case of multiple modules it looses the location.

**Reply by TheBroach** [reply]
Thanks for the pointers Clement!
If @geoand and @glefloch don't have any leads, maybe they can tag someone else to take a look?

**Reply by cescoffier** [reply]
Maybe @glefloch ?

**Comment by mschorsch** [comment]
At first sight i would say you have to configure the `protobuf-gradle-plugin` in "driven-adapters" so that `protoc` is fetched from maven central. See https://github.com/google/protobuf-gradle-plugin#locate-external-executables

**Reply by TheBroach** [reply]
I was under the impression the Quarkus Gradle Plugin does this for me?

**Reply by mschorsch** [reply]
You're right but why you have added the `protobuf-gradle-plugin`?

**Reply by mschorsch** [reply]
Could you try to remove the protobuf-gradle-plugin?

**Reply by TheBroach** [reply]
Ooooh. My bad. I added it when I tried out some suggestions from Github Copilot, but it hallucinated quite a bit... Forgot to remove it before uploading this MCVE, sorry!
It actually builds for me now! Is it possible for you to download and try to build?

**Reply by TheBroach** [reply]
Edit: It is building, but I can't see any generated gRPC classes in any build directory?

**Reply by mschorsch** [reply]
@TheBroach: You need to configure the Quarkus protobuf generation in module `application`. See https://quarkus.io/guides/grpc-generation-reference#configuring-grpc-code-generation-for-dependencies

**Reply by mschorsch** [reply, marked as answer]
If you really want to have the generated classes in the module `driven-adapters` then I would recommend you to generate the classes with the `protobuf-gradle-plugin` instead of `quarkus-grpc`.

**Reply by TheBroach** [reply]
Ah okay. Yeah, seems that I might be forced to abandon the Quarkus extension in this case. Will try with the vanilla Protobuf Gradle Plugin. Thank you so much for your assistance!
